app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt

```diff
@@ -69,5 +78,39 @@ class DashboardViewModel(
             .addTo(compositeDisposable)
     }
 
+    fun getPinUpdate() {
+        val prefData = appRepository.getPrefCountry()
+        if (prefData == null) {
+            _pinData.value = null
+        } else {
+            appRepository
+                .confirmed()
+                .map { stream -> stream.firstOrNull() { if (it.provinceState != null) it.provinceState == prefData.provinceState else it.countryRegion == prefData.countryRegion } }
+                .observeOn(schedulerProvider.ui())
+                .doOnSubscribe { _pinData.postValue(prefData) }
+                .subscribe({
+                    _pinData.postValue(it)
+                }, {
+                    _errorMessage.postValue(Constant.ERROR_MESSAGE)
+                })
+                .addTo(compositeDisposable)
+        }
+    }
+
+    fun getCountry(id: String) {
+        appRepository.country(id)
+            .observeOn(schedulerProvider.ui())
+            .doOnSubscribe {
+                appRepository.getCacheCountry(id)?.let { data ->
+                    _countryData.postValue(data)
+                }
+            }
+            .subscribe({
+                _countryData.postValue(it)
+            }, {
+                _errorMessage.postValue(Constant.ERROR_MESSAGE)
+            })
+            .addTo(compositeDisposable)
+    }
```

Comment:
can you create the test case for these ones, please?

Reply:
done